lib/server/services/qualityControlFlag/QcFlagService.js

```diff
@@ -418,19 +445,27 @@ class QcFlagService {
             const removedFlag = await QcFlagRepository.removeOne({ where: { id } });
 
             // Reconstruct flags ineffectiveness periods
-            const commonDataOrSimulationPassInclude = [ // QC flag can be associated with only one data pass or only one simulation pass
-                qcFlag.dataPasses?.length > 0
-                    ? { association: 'dataPasses', where: { id: qcFlag.dataPasses[0].id } }
-                    : { association: 'simulationPasses', where: { id: qcFlag.simulationPasses[0].id } },
-            ];
+            const flagIncludes = [];
+            let synchronousQcWhereClause = null;
+            // QC flag can be associated with only one data pass or only one simulation pass or be synchronous
+            if (qcFlag.dataPasses?.[0]?.id) {
```

Comment:
I do not understand here. 
First, if qcFlag can have only one dataPass, why is this property an array?
Second, why not simply checking the length as before?

Reply:
it's array, becuase there is relation many-to-many. It can be length